inference-engine/tests/functional/plugin/shared/src/single_layer_tests/pad.cpp

```diff
@@ -0,0 +1,78 @@
+// Copyright (C) 2020 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include <tuple>
+#include <vector>
+#include <string>
+#include <memory>
+#include <functional>
+
+#include "single_layer_tests/pad.hpp"
+
+
+namespace LayerTestsDefinitions {
+
+std::string PadLayerTest::getTestCaseName(testing::TestParamInfo<padLayerTestParamsSet> obj) {
+    InferenceEngine::Precision netPrecision;
+    InferenceEngine::SizeVector inputShapes, padsBegin, padsEnd;
+    ngraph::helpers::PadMode padMode;
+    float argPadValue;
+    std::string targetDevice;
+    std::tie(padsBegin, padsEnd, argPadValue, padMode, netPrecision, inputShapes, targetDevice) = obj.param;
+
+    std::ostringstream result;
+    result << "IS=" << CommonTestUtils::vec2str(inputShapes) << "_";
+    result << "padsBegin=" << CommonTestUtils::vec2str(padsBegin) << "_";
+    result << "padsEnd=" << CommonTestUtils::vec2str(padsEnd) << "_";
+    if (padMode == ngraph::helpers::PadMode::CONSTANT) {
+        result << "Value=" << argPadValue << "_";
+    }
+    result << "PadMode=" << padMode << "_";
+    result << "netPRC=" << netPrecision.name() << "_";
+    result << "targetDevice=" << targetDevice;
+    return result.str();
+}
+
+void PadLayerTest::SetUp() {
+    InferenceEngine::SizeVector inputShape, padsBegin, padsEnd;
+    float argPadValue;
+    ngraph::helpers::PadMode padMode;
+    InferenceEngine::Precision netPrecision;
+    std::tie(padsBegin, padsEnd, argPadValue, padMode, netPrecision, inputShape, targetDevice) = this->GetParam();
+    ngraph::op::PadMode pad_mode;
+    switch (padMode) {
+    case ngraph::helpers::PadMode::CONSTANT:
+        pad_mode = ngraph::op::PadMode::CONSTANT;
+        break;
+    case ngraph::helpers::PadMode::EDGE:
+        pad_mode = ngraph::op::PadMode::EDGE;
+        break;
+    case ngraph::helpers::PadMode::REFLECT:
+        pad_mode = ngraph::op::PadMode::REFLECT;
+        break;
+    case ngraph::helpers::PadMode::SYMMETRIC:
+        pad_mode = ngraph::op::PadMode::SYMMETRIC;
+        break;
+    default:
+        throw std::runtime_error("Can't create layer for this pad mode");
+    }
+
+    auto ngPrc = FuncTestUtils::PrecisionUtils::convertIE2nGraphPrc(netPrecision);
+    auto params = ngraph::builder::makeParams(ngPrc, {inputShape});
+    auto paramOuts = ngraph::helpers::convert2OutputVector(
+            ngraph::helpers::castOps2Nodes<ngraph::opset3::Parameter>(params));
+    auto pads_begin = std::make_shared<ngraph::opset3::Constant>(ngraph::element::i64, ngraph::Shape{padsBegin.size()}, padsBegin.data());
+    auto pads_end = std::make_shared<ngraph::opset3::Constant>(ngraph::element::i64, ngraph::Shape{padsEnd.size()}, padsEnd.data());
+
+    auto arg_pad_value = std::make_shared<ngraph::opset3::Constant>(ngPrc, ngraph::Shape{}, &argPadValue);
+    auto pad = std::make_shared<ngraph::opset3::Pad>(paramOuts[0], pads_begin, pads_end, arg_pad_value, pad_mode);
```

Comment:
Please, add the builder for this op. Looks like you move switch to the builder. I think this changes will be more elegant